src/linker/Linker/TypeNameResolver.cs

```diff
@@ -13,31 +13,35 @@ public TypeNameResolver (LinkContext context)
 			_context = context;
 		}
 
-		public TypeReference ResolveTypeName (string typeNameString)
+		public (AssemblyDefinition Assembly, TypeReference Type) ResolveTypeName (string typeNameString)
```

Comment:
I'd use `public TypeReference ResolveTypeName (string typeNameString, out AssemblyDefinition typeAssembly)` so the callers can use discard to indicate that the value is not relevant